### DISCUSSION TITLE
Required and optional user inputs, and legends as instructions

### DISCUSSION THREAD

**Discussion by mbgower** [w3c/wcag#3622] [Q&A]
In a recent PR for , as well as recent issues concerning the wording of the normative language for Labels or Instructions, questions have been raised about what is a sufficient/necessary use of legends to indicate required or optional user inputs.

This discussion will explore this notion, both in regard to WCAG 2 existing guidance and potential future 3.0 considerations.

https://github.com/w3c/wcag/issues/3553
https://github.com/w3c/wcag/pull/1792#issuecomment-1854339304

**Comment by bruce-usab** [comment]
@mraccess77 wrote:
> The proposed response that required fields don't need to be indicated seems to contradict previous advice from this group that fields which contain both optional and required fields would need to indicate either required or optional fields and not rely on form submission to provide that feedback unless they were something like a username/password field.

But where is the normative SC text?  It could not be requirement in all instances, since a field being required might not be determinable until the form is submitted.  Username / password gets an exception just because it is common, but that is not part of the SC text either.

**Comment by mraccess77** [comment]
While I don't think it is always required - the group has previously discussed situations where indicating required fields could be considered required.  For example, when the require state is known and where there are both required and optional fields. There has been numerous discussion on this and whether explanation of asterisks is required and I believe when consensus is reached we need to very clearly document what is and is not required as there is not consistency between those reporting findings on this topic before form validation.

**Comment by bruce-usab** [comment]
> While I don't think it is always required...

I am of the opinion that [detlev's proposed response](https://github.com/w3c/wcag/issues/3553#issuecomment-1843208880) well addresses the nuance.

**Comment by mbgower** [comment]
Note that an [official working group response](https://github.com/w3c/wcag/issues/3553#issuecomment-2022967832) was published, closing #3553

**Reply by pete-murphy** [reply]
> Note that an [official working group response](https://github.com/w3c/wcag/issues/3553#issuecomment-2022967832) was published, closing #3553

How does [Required Fields - Easy Checks](https://www.w3.org/WAI/test-evaluate/easy-checks/required-fields/) relate to this? There seems to be conflicting guidance between:

> If some or all fields are required for submission, there can be different ways of indicating that, and the SC does not prescribe a particular method to do it. It could be done, for example, by ... indicating optional fields (by implication the other fields would be required),

and

> If fields are marked as ‘optional’ and not ‘required’ then check to make sure there is a message to say that all fields are required unless marked otherwise

Following the former guidance, I might leave out any top-level message or instruction that fields not marked as optional are required, but the latter guidance seems to expect it.

**Reply by patrickhlauke** [reply]
> How does [Required Fields - Easy Checks](https://www.w3.org/WAI/test-evaluate/easy-checks/required-fields/) relate to this? There seems to be conflicting guidance

that easy check (in draft at the moment) goes beyond the normative requirement, I'd argue, and should be fixed - or it should clearly say that this verges on best practice advice (cc @iadawn)

again, i'll note that context also plays a role. there are types of forms, like username/password ones, where it's self-evident that both input fields are required. a "sign up to our newsletter" form with a single email address input ... it's clear that that single input is required (otherwise, what would you even submit when pressing "sign up" button that submits the form).

there's nuance in real-world content, and WCAG itself does not prescribe specifically that required fields MUST ALWAYS be EXPLICITLY labelled as such.
